langchain4j-core/src/main/java/dev/langchain4j/store/embedding/EmbeddingStore.java

```diff
@@ -55,11 +57,12 @@ public interface EmbeddingStore<Embedded> {
     /**
      * Adds multiple embeddings and their corresponding contents that have been embedded to the store.
      *
+     * @param ids A list of embeddings ids associated with the added embeddings.
      * @param embeddings A list of embeddings to be added to the store.
      * @param embedded   A list of original contents that were embedded.
-     * @return A list of auto-generated IDs associated with the added embeddings.
+     * @return A list of IDs associated with the added embeddings.
      */
-    List<String> addAll(List<Embedding> embeddings, List<Embedded> embedded);
+    void addAll(List<String> ids, List<Embedding> embeddings, List<Embedded> embedded);
```

Comment:
Just in case: this is not **_very_** critical, but if there is a way to avoid breaking chnages, it would be great. If it requires tonnes of manual changes, I guess we can live with it

Reply:
it's a new method, And I implemented it everywhere

Reply:
Not everywhere, there are implementations of `EmbeddingStore` interface that live outside of this repository

Reply:
I can do it, but I'm not sure that it is a good desicion.

It will compile every time, and not implemented - not very good.

Reply:
done/ but don't like it

Reply:
@VictorZZZZ sorry, I forgot to sumbit my comments :/

Reply:
don't like 'default' because we are doing a new version of library, so it's a new method in our EmbeddingStore interface and all required implementations in library we made. 
We don't know who will use and it's OK, we shoudn't know this.
If somebody wants to use a new version, they must be ready to implement required methods.

And here it turns out that by adding a `default`, we are accommodating them so as not to break anything for them. But it should be the other way around.

Reply:
@VictorZZZZ I see, thank you for the explanation. I see it differently. This library is used by many people, and it is our duty to make the experience for them as seamless as possible. It does not cost us much to avoid breaking changes, so I see no reason to break anything for our users, especially on purpose. Backward compatibility is a thing, especially in Java world.

Reply:
@langchain4j 
It's a new method. There is no anything about "Backward compatibility"

Reply:
If you introduce a new non-default method to the interface, all implementations of this interface have to be updated, otherwise it will not compile

Reply:
it's a new version of core library. User must be ready for this. 
Backward compatibility will not be affected.

Reply:
This would be OK if we were to release a major version update, but we are far from it. So let's keep minor versions without too many breaking changes.

Reply:
I didn't think about it in minor/major versions projection.

But there are some changes in already existed method `addAll(List<Embedding> embeddings, List<Embedded> embedded)`

now it uses new method `addAll(List<String> ids, List<Embedding> embeddings, List<Embedded> embedded)` - which is throws exception by default.

Reply:
Ah, so you mean that other implementations of `EmbeddingStore` that we do not control will throw an exception when `addAll(List<Embedding> embeddings, List<Embedded> embedded)` is called?
This is not nice, it could by decoupling `addAll(List<Embedding> embeddings, List<Embedded> embedded)` from the new method. WIll this be too much of changes?

Reply:
Yes `EmbeddingStores` we don't control will be broken if there won't be implementation of new method.

Decoupling is not hard. But it's not correct.
That's why i'd like to make a new method required to implement.

Reply:
If decoupling is not hard, could you please do it?

I see no reason why existing code that uses `EmbeddingStore.addAll(embeddings, segments)` or `EmbeddingStore` in general should break when updating to the next LC4j version.

It should only fail if some (unknown to us) `EmbeddingStore` implementation does not implement the new default `addAll(ids, embeddings, segments)` method, but someone calls it. In all other cases there should be no compilation or runtime errors.

Reply:
@langchain4j, I have don decoupling in the implementations

Reply:
...

Reply:
OK.
Reverted my last commit